jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js

```diff
@@ -160,20 +159,23 @@ class Charmbrowser extends React.Component {
       case 'entity-details':
         // TODO frankban: do we still really need this?
         const id = currentState.store || `~${currentState.user}`;
+        const { charmstore } = this.props;
         activeChild = (
           <EntityDetails
             acl={this.props.acl}
             addNotification={this.props.addNotification}
-            apiUrl={this.props.apiUrl}
             changeState={changeState}
+            charmstore={shapeup.addReshape({
```

Comment:
`charmstore={this.props.charmstore.reshape(EntityDetails.propTypes.api)}` ?

Reply:
Thanks, good suggestion.